pkg/cmd/experimental/bundlesecret/bundle-secret.go

```diff
@@ -0,0 +1,114 @@
+package bundlesecret
+
+//export OPENSHIFTCONFIG=./openshift.local.certificates/admin/.kubeconfig
+import (
+	"fmt"
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	kcmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/spf13/cobra"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+const MaxSecretSize = 1 * 1024 * 1024
+
+func NewCmdBundleSecret(f *clientcmd.Factory, parentName, name string, out io.Writer) *cobra.Command {
+	var sourcelist util.StringList
+	cmd := &cobra.Command{
+		Use:   "bundle-secret NAME -f SOURCENAME",
+		Short: "bundle data from a source list to load kubernetes secret",
+		Long:  "bundle data from a source list to load kubernetes secret",
+		Run: func(cmd *cobra.Command, args []string) {
+			var err error
+			if len(args) == 0 {
+				err = fmt.Errorf("Must specify NAME as an argument")
+				checkErr(err, out)
+			}
+			objname := args[0]
+			if len(sourcelist) == 0 {
+				err = fmt.Errorf("Must specify files and/or directories to gather data from.")
+				checkErr(err, out)
+			}
+
+			secretdata := make(map[string][]byte)
+			for _, sourceitem := range sourcelist {
+				sourceitem = strings.TrimSuffix(sourceitem, "/")
+				info, err := os.Stat(sourceitem)
+				checkErr(err, out)
+
+				if info.IsDir() {
+					filelist, err := ioutil.ReadDir(sourceitem)
+					checkErr(err, out)
+					for _, item := range filelist {
+						if item.IsDir() {
+							return
+						} else {
+							file := fmt.Sprint(sourceitem, "/", item.Name())
+
+							readValidFiles(file, item.Name(), secretdata, out)
+						}
+					}
+				} else {
+					file := sourceitem
+					readValidFiles(file, info.Name(), secretdata, out)
+				}
+			}
+			secretobj := kapi.Secret{
+				ObjectMeta: kapi.ObjectMeta{Name: objname},
+				Data:       secretdata,
+			}
+			totalsize := 0
```

Comment:
i would leave enforcement of total secret size to the server and eliminate this.